.buildkite/tests.sh

```diff
@@ -80,8 +80,8 @@ done
 
 # setup secrets
 set +x # dont echo secrets
-echo "DOCKERHUB_MIRROR_USER=$($earthly secret --org earthly-technologies --project core get -n dockerhub-mirror/user)" > .secret
-echo "DOCKERHUB_MIRROR_PASS=$($earthly secret --org earthly-technologies --project core get -n dockerhub-mirror/pass)" >> .secret
+echo "DOCKERHUB_MIRROR_USER=$($earthly secret --org earthly-technologies --project core get -n dockerhub-mirror/user || kill $$)" > .secret
+echo "DOCKERHUB_MIRROR_PASS=$($earthly secret --org earthly-technologies --project core get -n dockerhub-mirror/pass || kill $$)" >> .secret
```

Comment:
nit: 
just in case it might look cleaner (not sure what happens after kill)
could we instead check the status of  $? and exit on non zero?

Reply:
> could we instead check the status of $? and exit on non zero?

It could potentially be done with named pipes, but that seems rather overkill. I'm not sure I understand your suggestion -- it's unclear how this could be done via a command substitution.

In this case of the kill -- we'll see the error message from earthly written to stderr, followed by a "terminated" message.

Reply:
sounds good.
my suggestion was to do:
```
if [ $? != 0] then
   exit 1
end if
```
but I agree it's likely an overkill and will make the script more verbose.

Reply:
It's still unclear how that would work with a command substitution.

An `exit 1` will not propagate to the caller.